### ISSUE TITLE
Undefined binaries in flow: ('H-rich_Core_H_burning', 'H-rich_Core_H_burning', 'RLO2', 'oRLO2')

### ISSUE THREAD

**Issue by celiotine** [POSYDON-code/POSYDON-public#419]
Binaries with the parameters ('H-rich_Core_H_burning', 'H-rich_Core_H_burning', 'RLO2', 'oRLO2') are failing because they are not handled in the flow. An example of one such binary:

```
ini_filepath_IF = "/projects/b1119/cliotine/pulsar_dev/pulsar_pops/runs/oct_2024/Sukhbold+16/grid_reruns_final/params.ini"
sim_kwargs_IF = simprop_kwargs_from_ini(ini_filepath_IF)
sim_prop_IF = SimulationProperties(**sim_kwargs_IF)
sim_prop_IF.load_steps(verbose=True)

star_1 = SingleStar(**{'mass':   6.263565862272097, 'state': 'H-rich_Core_H_burning',
                'natal_kick_array': [None, None, None, None]})
star_2 = SingleStar(**{'mass': 4.042250364920023, 'state': 'H-rich_Core_H_burning',
                'natal_kick_array': [None, None, None, None]})
binary_IF = BinaryStar(star_1, star_2, **{'time': 0.0,  'state': 'detached',  'event': 'ZAMS', 
                        'orbital_period':  2.3130949924692237, 'eccentricity': 0.0}, properties = sim_prop_IF)
binary_IF.evolve()
```
<img width="623" alt="Screen Shot 2024-10-22 at 9 38 00 AM" src="https://github.com/user-attachments/assets/283acebd-74cf-4f8b-9eaa-001ea0cf0c24">

**Comment by celiotine** [comment]
Per discussion 10/24: this is most likely a bug with labeling the star type, should go to CC1 and form WD?

**Comment by celiotine** [comment]
These binaries should be handled by PR #430.

**Comment by celiotine** [comment]
In addition, I am wondering if something wonky is happening with the interpolation for these binaries? It is still unclear to me why they are entering RLO after HMS-HMS to begin with. This is the interpolation information for the binary above:

<img width="275" alt="Screen Shot 2024-11-19 at 2 41 35 PM" src="https://github.com/user-attachments/assets/bfeb7554-6158-4a31-b46e-78bfe5b13549">
 
So not sure if these systems should be entering CE?

**Comment by maxbriel** [comment]
@celiotine Yep. That looks like another classifier inconsistency to me. I've seen similar things in this issue #402

**Comment by celiotine** [comment]
From discussion on 11/21: We think this is an interpolator issue, as the binary interpolation class and MT history do not match (former is unstable, latter is stable). @philipp-rajah and @mkruckow plan to discuss possible solutions in more detail.

For now, we agreed we can create a separate POSYDON error to handle these binaries, but keep the issue open until the underlying issue is resolved.

**Comment by celiotine** [comment]
PR #430 was merged, which includes a ClassificationError catch for these binaries in the detached step.
I am leaving the issue open since the underlying problem still exists.

**Comment by astroJeff** [comment]
@celiotine just to check, is this a still-open issue because the ClassificationError check doesn't check all cases of inconsistencies?

**Comment by mkruckow** [comment]
It's more about that we only get an error instead of having a real solution to let the system evolve correctly.
I see two solution options:

1. Disallow the misclassification of the interpolator (preferred solution and hopefully included in the rewriting of the interpolators)
2. Use a HMS-HMS_RLO gird (could be done quickly, but it would request a larger v2 dataset.)

**Comment by sgossage** [comment]
Linked this to [PR#615](https://github.com/POSYDON-code/POSYDON/pull/615), the interpolator rewrite. We should check whether a solution may be found there, or else I'd advocate for the HMS-HMS_RLO grid anyways in future versions of POSYDON.
